### ISSUE TITLE
Consider EOL'ing this project in favour of AWS SSO?

### ISSUE THREAD

**Issue by brainstorm** [cevoaustralia/aws-google-auth#190]
We recently migrated and use AWS SSO and [yawsso](https://github.com/victorskl/yawsso/) as a CLI v1 backwards compatibility workaround. Other than this minor temporary patch (due to an unfixed AWS Go SDK issue) it has been more reliable than this @cevoaustralia's (and others) web scraping approaches.

Please don't get me wrong, this project has been instrumental for a while at our organization but I suspect that the scraping limitations (and reliability) will only get worse over time, driving folks away from deploying reliable SSO systems.

In other words: what's the current value proposition of this project vs AWS SSO? Any additional features that I'm not aware of? Happy to be wrong about this!

**Comment by zeroaltitude** [comment]
My question @brainstorm is, how does this solve getting the Google SSO dance automated? The value of projects like aws-google-auth is actually handing the Google side, and that is the side that to this day remains tricky. How do you accomplish that with yawsso? I read the page, and don't understand.

**Comment by brainstorm** [comment]
@zeroaltitude Setting up AWS SSO with Google is fairly straightforward with SAML, here's a step by step guide (on of many):

https://deductivelabs.com/blog/aws/amazon-web-services-sso-authentication-with-google-gsuite/

Yawsso only covers the "last mile" tools like terraform or CDK that do not yet support AWS SSO but just plain AWS CLI v1, as mentioned above w/ the AWS SDK Go pending issue.

**Comment by zeroaltitude** [comment]
I think I may have been unclear. I have already set up AWS to use federated login from Google SSO as per your link. Now, I want to generate command line tools for developers so that, using their Google creds, they can use AWS APIs exposed through Boto3. I use a tool that derives from the current project to do that. It allows me to automate a relatively seamless experience for a user wanting to use client tools to make AWS calls, having only authenticated with their Google identity as set up using SSO GSuite.

So that's the role that the current project (a derivative of it) plays for me. You have suggested EOLing the project. You have suggested that I could just use Yawsso. I don't see how I can do that using Yawsso. That is, I do not see how to use Yawsso to 1) integrate in a command line setting, in order to 2) allow a developer to provide GSuite credentials, so that they can then 3) have their AWS credentials file set up correctly so they can use boto tools from within python commands from the commandline.

If Yawsso does that, that's fantastic. I would like to understand how. If it does not, then I think EOLing this project in favor of it would be premature, as this project allows me to do just that.

**Comment by zeroaltitude** [comment]
Oh wait, I'm starting to see the light. I didn't set up AWS SSO, I set up a custom SAML provider in each account.

OK, so my question actually is: back when I did this, it didn't seem like AWS SSO would easily allow me to get the kinds of creds I would need in order to run command line tools such as python programs such as boto. Was I wrong, or, has this been enhanced?

**Comment by zeroaltitude** [comment]
Drat. I'm so sorry for the spam. I lost you again. The *Google AWS SSO app* does use a SAML identity provider on AWS IAM. So my set up is exactly as your link suggests, but still, my question about how to automate command line tools stands.

What confused me is that AWS now has it's OWN tool called AWS SSO. It simplifies this whole process. But it only works with Active Directory and its ilk. Though, it does support OKTA, which can then be used to bridge out somewhere else. But that's too complicated.

What I want is your link, plus command line tools. How do I do that without this project?

**Comment by zeroaltitude** [comment]
Oh, huh, nevermind, found it: https://medium.com/@io_78824/introduction-8a480b2df991

Less than 6 months old. This looks promising.

**Comment by brainstorm** [comment]
Yeah, SCIM is still lacking, that's together with having to use the yawsso "bridge", the only significant drawback, other than that, works a treat!

**Comment by brokenthumbs** [comment]
Referencing a ticket here on the aws-cli repo, to track the progress.

https://github.com/aws/aws-cli/issues/4784

**Comment by stevemac007** [comment]
Thanks for the question - something we should constantly be looking at.  I'd love to be able to retire this tool, especially given how lax I've been during 2020 responding to issues and keeping it updated.  But there does still seem to be a demand for keeping this alive in the meantime.

I'm aiming to get a run at some uplift here over the next week or so to resolve any key blockers - the first of which is to move off Travis CI to GitHub Actions - and look into the remaining PR's.

**Comment by brainstorm** [comment]
At least put a reference/link to the very beginning of the README pointing to this issue?

In our experience, our transition to AWS SSO has been the difference between tons of bad UX reports with `aws-google-auth` and 0 related with AWS SSO... inform new users!

**Comment by brainstorm** [comment]
Btw, support for CLIv2 on AWS Terraform provider has been merged: https://github.com/hashicorp/terraform-provider-aws/issues/10851#issuecomment-769434408

The AWS Go SDK is supporting it now, so that was pretty much the last inconvenience milestone to fully adopt SSO I reckon: https://github.com/hashicorp/terraform-provider-aws/pull/17340/files
